drivers/timer/nrf_rtc_timer.c

```diff
@@ -83,7 +85,7 @@ int z_clock_driver_init(struct device *device)
 
 	ARG_UNUSED(device);
 
-	clock = device_get_binding(DT_INST_0_NORDIC_NRF_CLOCK_LABEL);
+	clock = device_get_binding(DT_INST_LABEL(0));
```

Comment:
Same story as in `counter_nrfx_rtc`, this is not a driver for "nordic,nrf-clock" devices.
```suggestion
	clock = device_get_binding(DT_PROP(DT_INST(0, nordic_nrf_clock), label));
```